### ISSUE TITLE
Error response from daemon: error while mounting volume

### ISSUE THREAD

**Issue by brichards-zb** [docksal/docksal#1771]
**Description**

I'm trying to set up docksal on an M1 mac and I keep running into issues when I run `fin init`.

I'm running v 4.18.0 of Docker Desktop.

I'm getting the error:
`Error response from daemon: error while mounting volume '/var/lib/docker/volumes/myproject_project_root/_data': failed to mount local volume: mount :/Users/billrichards/Projects/myproject:/var/lib/docker/volumes/myproject_project_root/_data, flags: 0xc00, data: addr=192.168.64.1,vers=3,nolock,noacl,nocto,actimeo=1,wsize=32768,rsize=32768: connection refuse
`

My project is in the Projects folder of my home directory, which is the Docksal recommended directory and looking under Resources -> File Sharing I can see that /Users is available for File Sharing.

Sometimes it doesn't even get as far as failing. Sometimes it just hangs on some random step of the process. 

**Steps to reproduce the issue:**

1. run `fin init` from my project directory

**Describe the results you received:**

Error response from daemon: error while mounting volume '/var/lib/docker/volumes/myproject_project_root/_data': failed to mount local volume: mount :/Users/billrichards/Projects/myproject:/var/lib/docker/volumes/myproject_project_root/_data, flags: 0xc00, data: addr=192.168.64.1,vers=3,nolock,noacl,nocto,actimeo=1,wsize=32768,rsize=32768: connection refuse

**Describe the results you expected:**

I expected docksal to start.

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
 WARNING:  '.docksal' base domain and local DNS resolver are deprecated
           Please update your Docksal and project configuration
           See https://docs.docksal.io/stack/configuration-variables/#docksal-dns-disabled
---------------------
COMPOSE_PROJECT_NAME_SAFE: myproject
COMPOSE_FILE:
/Users/billrichards/.docksal/stacks/volumes-nfs.yml
/Users/billrichards/.docksal/stacks/stack-default.yml
/Users/billrichards/Projects/myproject/.docksal/docksal.yml
ENV_FILE:
/Users/billrichards/Projects/myproject/.docksal/docksal.env

PROJECT_ROOT: /Users/billrichards/Projects/myproject
DOCROOT: docroot
VIRTUAL_HOST: myproject.docksal
VIRTUAL_HOST_ALIASES: *.myproject.docksal
IP: 192.168.64.100

MySQL endpoint:
Public URL:

Docker Compose configuration
---------------------
name: myproject
services:
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      BLACKFIRE_CLIENT_ID: null
      BLACKFIRE_CLIENT_TOKEN: null
      COMPOSER_ALLOW_XDEBUG: "0"
      COMPOSER_DEFAULT_VERSION: null
      COMPOSER_DISABLE_XDEBUG_WARN: "0"
      DOCROOT: docroot
      DRUSH_ALLOW_XDEBUG: "0"
      DRUSH_OPTIONS_URI: myproject.docksal
      GIT_USER_EMAIL: ""
      GIT_USER_NAME: ""
      HOST_GID: "20"
      HOST_UID: "502"
      MYSQL_DATABASE: default
      MYSQL_HOST: db
      MYSQL_PASSWORD: user
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
      PHP_IDE_CONFIG: null
      SECRET_ACQUIA_CLI_KEY: null
      SECRET_ACQUIA_CLI_SECRET: null
      SECRET_PLATFORMSH_CLI_TOKEN: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      SSH_AUTH_SOCK: /.ssh-agent/proxy-socket
      VIRTUAL_HOST: myproject.docksal
      XDEBUG_CONFIG: client_host=192.168.64.1
      XDEBUG_ENABLED: "0"
    extends:
      file: /Users/billrichards/.docksal/stacks/services.yml
      service: cli
    hostname: cli
    healthcheck:
      interval: 10s
    image: docksal/cli:php8.1-3.3
    labels:
      io.docksal.shell: bash
      io.docksal.user: docker
    logging:
      options:
        max-file: "10"
        max-size: 1m
    networks:
      default: null
    volumes:
    - type: volume
      source: docksal_ssh_agent
      target: /.ssh-agent
      read_only: true
      volume: {}
    - type: volume
      source: cli_home
      target: /home/docker
      volume: {}
    - type: bind
      source: /tmp/.docksal/myproject
      target: /tmp/.docksal/myproject
      read_only: true
      bind:
        create_host_path: true
    - type: volume
      source: project_root
      target: /var/www
      volume:
        nocopy: true
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: null
      MYSQL_DATABASE: default
      MYSQL_INITDB_SKIP_TZINFO: null
      MYSQL_ONETIME_PASSWORD: null
      MYSQL_PASSWORD: user
      MYSQL_RANDOM_ROOT_PASSWORD: null
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    extends:
      file: /Users/billrichards/.docksal/stacks/services.yml
      service: mariadb
    hostname: db
    healthcheck:
      interval: 10s
    image: docksal/mariadb:10.6-1.3
    logging:
      options:
        max-file: "10"
        max-size: 1m
    networks:
      default: null
    ports:
    - mode: ingress
      target: 3306
      protocol: tcp
    volumes:
    - type: volume
      source: db_data
      target: /var/lib/mysql
      volume: {}
    - type: volume
      source: project_root
      target: /var/www
      read_only: true
      volume:
        nocopy: true
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
      APACHE_FCGI_HOST_PORT: cli:9000
    extends:
      file: /Users/billrichards/.docksal/stacks/services.yml
      service: apache
    hostname: web
    healthcheck:
      interval: 10s
    image: docksal/apache:2.4-2.5
    labels:
      io.docksal.cert-name: none
      io.docksal.permanent: "false"
      io.docksal.project-root: /Users/billrichards/Projects/myproject
      io.docksal.virtual-host: myproject.docksal,*.myproject.docksal,myproject.docksal.*
    logging:
      options:
        max-file: "10"
        max-size: 1m
    networks:
      default: null
    volumes:
    - type: volume
      source: project_root
      target: /var/www
      read_only: true
      volume:
        nocopy: true
networks:
  default:
    name: myproject_default
volumes:
  cli_home:
    name: myproject_cli_home
  db_data:
    name: myproject_db_data
  docksal_ssh_agent:
    name: docksal_ssh_agent
    external: true
  project_root:
    name: myproject_project_root
    driver: local
    driver_opts:
      device: :/Users/billrichards/Projects/myproject
      o: addr=192.168.64.1,vers=3,nolock,noacl,nocto,noatime,nodiratime,actimeo=1,wsize=32768,rsize=32768
      type: nfs
---------------------
```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
  
███  DOCKSAL
Docksal version: v1.18.2
fin version:     v1.113.2

███  OS
Darwin macOS 12.3
Darwin BRichards-MacBookPro 21.4.0 Darwin Kernel Version 21.4.0: Mon Feb 21 20:35:58 PST 2022; root:xnu-8020.101.4~2/RELEASE_ARM64_T6000 arm64

███  ENVIRONMENT
MODE : Docker Desktop
DOCKER_HOST : 

███  NFS
DOCKSAL_NFS_PATH : /Users

nfsd service is disabled
nfsd is not running

NFS EXPORTS
----------
cat: /etc/exports: No such file or directory
----------

showmount: Cannot retrieve info from host: localhost: RPC failed:: RPC: Timed out

███  DOCKER
Expected client version: 20.10.23
Expected server version: 20.10.23

Installed versions:

Client:
 Version:           20.10.23
 API version:       1.41
 Go version:        go1.18.10
 Git commit:        7155243
 Built:             Thu Jan 19 17:35:19 2023
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.18.0 (104112)
 Engine:
  Version:          20.10.24
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.19.7
  Git commit:       5d6db84
  Built:            Tue Apr  4 18:17:07 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.18
  GitCommit:        2456e983eb9e37e47538f59ea18f2043c9a73640
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

███  DOCKER COMPOSE
Expected version:  2.15.1
Installed version: 2.15.1

███  DOCKSAL: PROJECTS
project          STATUS    virtual host                                                                     project root
My project	    Created   myproject.docksal.site,*.myproject.docksal.site,myproject.docksal.site.*   /Users/billrichards/Projects/myproject

███  DOCKSAL: VIRTUAL HOSTS
pma-myproject.docksal.site

███  DOCKSAL: NETWORKING

DOCKSAL_IP: 192.168.64.100
DOCKSAL_HOST_IP: 192.168.64.1
DOCKSAL_VHOST_PROXY_IP: 0.0.0.0
DOCKSAL_DNS_IP: 0.0.0.0
DOCKSAL_DNS_DISABLED: 0
DOCKSAL_NO_DNS_RESOLVER: 0
DOCKSAL_DNS_UPSTREAM: 
DOCKSAL_DNS_DOMAIN: docksal

███  DOCKSAL: CONNECTIVITY

Host to 192.168.64.100: 	PASS
Container to 192.168.64.100: 	PASS
Container to 192.168.64.1: 	FAIL

Checking connectivity to http://dns-test.docksal...
Host: FAIL

Debug info:
----------
+ cat /etc/resolv.conf
+ grep 192.168.64.100
+ ping -c 1 -W 1 dns-test.docksal
PING dns-test.docksal (192.168.64.100): 56 data bytes
64 bytes from 192.168.64.100: icmp_seq=0 ttl=64 time=0.659 ms

--- dns-test.docksal ping statistics ---
1 packets transmitted, 1 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 0.659/0.659/0.659/0.000 ms
+ nslookup -timeout=1 dns-test.docksal 192.168.64.100
;; connection timed out; no servers could be reached

+ set +x
----------

Containers: FAIL

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID   IMAGE                     COMMAND                  CREATED          STATUS                    PORTS                                      NAMES
2edc58b99d01   phpmyadmin/phpmyadmin     "/docker-entrypoint.…"   13 minutes ago   Up 7 minutes              80/tcp                                     myproject_pma_1
9d003b5a95c8   docksal/ssh-agent:1.4     "docker-entrypoint.s…"   49 minutes ago   Up 49 minutes (healthy)                                              docksal-ssh-agent
ed8492ccf0e1   docksal/dns:1.2           "docker-entrypoint.s…"   50 minutes ago   Up 49 minutes (healthy)   0.0.0.0:53->53/udp                         docksal-dns
5f8150bae7e8   docksal/vhost-proxy:1.8   "docker-entrypoint.s…"   50 minutes ago   Up 49 minutes (healthy)   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   docksal-vhost-proxy

███  DOCKER: NETWORKS
NETWORK ID     NAME                     DRIVER    SCOPE
8709d4c2238c   _default                 bridge    local
70916b643990   bridge                   bridge    local
6608ddfb273b   host                     host      local
62ca089759ec   myproject_default        bridge    local
3a72c1256962   none                     null      local

███  DOCKER DESKTOP
EXPECTED VERSION: 4.17.0
DETECTED VERSION: 4.18.0

███  HDD Usage
Filesystem       Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk3s1s1  460Gi   22Gi  370Gi     6%  500632 3876621480    0%   /
devfs           201Ki  201Ki    0Bi   100%     696          0  100%   /dev
/dev/disk3s6    460Gi   20Ki  370Gi     1%       0 3876621480    0%   /System/Volumes/VM
/dev/disk3s2    460Gi  497Mi  370Gi     1%     841 3876621480    0%   /System/Volumes/Preboot
/dev/disk3s4    460Gi  633Mi  370Gi     1%     246 3876621480    0%   /System/Volumes/Update
/dev/disk1s2    500Mi  6.0Mi  480Mi     2%       1    4910600    0%   /System/Volumes/xarts
/dev/disk1s1    500Mi  7.4Mi  480Mi     2%      29    4910600    0%   /System/Volumes/iSCPreboot
/dev/disk1s3    500Mi  2.3Mi  480Mi     1%      49    4910600    0%   /System/Volumes/Hardware
/dev/disk3s5    460Gi   66Gi  370Gi    16%  548474 3876621480    0%   /System/Volumes/Data
map auto_home     0Bi    0Bi    0Bi   100%       0          0  100%   /System/Volumes/Data/home
/dev/disk3s1    460Gi   22Gi  370Gi     6%  502146 3876621480    0%   /System/Volumes/Update/mnt1
  ```

</details>

**Comment by lmakarov** [comment]
Please run `fin system start` and share the output here.

**Comment by brichards-zb** [comment]
`fin system start`
```Configuring network settings...
Configuring network settings...
Configuring NFS shares...
Password:
 ERROR:  There is an error in /etc/exports file
         Read through the error message(s) below and fix errors in /etc/exports manually
         Then run fin system reset
-----------------
exports:0: nfsd is not running, can't verify exports permissions
-----------------
 ERROR:  NFS sharing has failed
billrichards in ~/Projects/myproject [main] > 
```
I've seen this sometimes while attempting to run `fin init`.
If I do a `nfsd status`, it says that the service is disabled and is not running. And my /etc/exports file is empty again. Yesterday I updated it to match the one on the laptop that I am replacing. It had one line `/Users 127.0.0.1 192.168.64.1 -alldirs -maproot=0:0` in it. But it keeps disappearing. If I enable nfsd and recreate my /etc/exports, sometimes `fin init` gets further in the process. Yesterday I ran it and it hung after Importing our db.

It sometimes gets as far as appearing to log php errors and I can see the website pulling up properly, but it still isn't quite right. Usually when running `fin init` the running finishes and brings me back to the terminal prompt. Even when it gets as far as displaying the php errors, it never returns.

**Comment by lmakarov** [comment]
There is something finicky with this machine's nfsd service or network settings. See if this is helpful:
https://apple.stackexchange.com/questions/302527/issues-sharing-folder-through-nfs-rpc-unable-to-send-errno-bad-file-descri

**Comment by brichards-zb** [comment]
> https://apple.stackexchange.com/questions/302527/issues-sharing-folder-through-nfs-rpc-unable-to-send-errno-bad-file-descri

No. My /etc/hosts matched theirs perfectly. 

But it's doing the thing again where it appears to start and work. I ran `fin init` and while it never finished, it appears to be working. In the terminal where I ran `fin int`, after importing the DB, it started showing the error log in the terminal. Its appears to have hung, but after walking away and coming back an hour later, it seems to have successfully finished.

I rebooted my machine and ran 'fin start' and everything ran fine.

I did upgrade my system to Ventura. Maybe that helped. I don't know, but in any case, it seems to be working now. 

Thank you.
